src/app/tabs/compileTab/compilerContainer.js

```diff
@@ -124,7 +124,7 @@ class CompilerContainer {
         const pragmaStr = pragmaArr[0].replace('pragma solidity', '').trim()
         const pragma = pragmaStr.substring(0, pragmaStr.length - 1)
         const fixedVersions = this.data.allversions.filter(obj => !obj.prerelease).map(obj => obj.version)
-        if (!semver.satisfies(this._retrieveVersion(), pragma)) {
+        if (fixedVersions.includes(this.data.selectedVersion) && !semver.satisfies(this._retrieveVersion(), pragma)) {
```

Comment:
I would create a variable or function that describe this statement, because I cannot understand it by reading the code only : 
```typescript
const is**** = fixedVersions.includes(this.data.selectedVersion) && !semver.satisfies(this._retrieveVersion(), pragma)
if (is***) {...}
```